clustered/client/src/main/java/org/ehcache/clustered/client/internal/loaderwriter/ClusteredLoaderWriterStore.java

```diff
@@ -309,4 +318,39 @@ private V loadFromLoaderWriter(K key) {
     return null;
   }
 
+  /**
+   * Provider of {@link ClusteredLoaderWriterStore} instances.
+   */
+  @ServiceDependencies({ TimeSourceService.class, ClusteringService.class})
+  public static class Provider extends ClusteredStore.Provider {
+    @Override
+    protected <K, V> ClusteredStore<K, V> createStore(Configuration<K, V> storeConfig,
+                                                      OperationsCodec<K, V> codec,
+                                                      ChainResolver<K, V> resolver,
+                                                      TimeSource timeSource,
+                                                      boolean useLoaderInAtomics,
+                                                      Object[] serviceConfigs) {
+      return new ClusteredLoaderWriterStore<>(storeConfig, codec, resolver, timeSource,
+                                              storeConfig.getCacheLoaderWriter(), useLoaderInAtomics);
+    }
+
+    @Override
+    public int rank(Set<ResourceType<?>> resourceTypes, Collection<ServiceConfiguration<?>> serviceConfigs) {
+      int parentRank = super.rank(resourceTypes, serviceConfigs);
+      if (parentRank == 0 || serviceConfigs.stream().noneMatch(CacheLoaderWriterConfiguration.class::isInstance)) {
+        return 0;
+      }
+      return parentRank + 1;
+    }
+
+    @Override
+    public int rankAuthority(ResourceType<?> authorityResource, Collection<ServiceConfiguration<?>> serviceConfigs) {
+      int parentRank = super.rankAuthority(authorityResource, serviceConfigs);
```

Comment:
We need to add test coverage for this ranking change for authoratative tier.

Reply:
added